M0090IDE-3982 document resolve in workspace (#443)
MicroEJ/docs: SDKUserGuide/resolve.rst

> .. resolve:

Reviewer: ```suggestion
.. resolveInWorkspace:
```

Resolve in Workspace
====================

> By default, the ``IvyDE workspace dependencies resolution`` is enabled in the SDK. When resolving a module's dependencies, if a dependency is opened in

Author: Done

> the same workspace as the module, the workspace resolution will link both the module and its dependency instead of downloading the published artifact, provided that the dependency's project has the same version as the one required by the module. 

Reviewer: ```suggestion
the same workspace as the module, the project is directly used as a dependency instead of downloading the published artifact, provided that the dependency's project has the same version as the one required by the module. 
```

> the same workspace as the module, the workspace resolution will link both the module and its dependency instead of downloading the published artifact, provided that the dependency's project has the same version as the one required by the module. 

Reviewer: We should add a screenshot of a project used as a dependency of a project in the SDK.

Author: Done

For example, suppose that the workspace contains a module A which has the following dependency :

::

    <dependency org="com.mycompany" name="mylib" rev="1.0.0" />

> If the ``mylib`` dependency is located in the workspace and its version is ``1.0.0``, it will be linked with the module A, else the published artifact with the right version will be downloaded.

Reviewer: ```suggestion
If the ``mylib`` project is located in the workspace and its version is ``1.0.0``, it is used for compilation and execution. Otherwise the published artifact is downloaded from the artifact repository.
```

> To avoid a dependency to be resolved in the workspace, simply close the corresponding project or remove it from the workspace. The workspace resolution can also be only disabled for the Foundation Libraries. To do so, go to ``Window > Preferences > MicroEJ > Settings`` and uncheck the ``Resolve Foundation Library in workspace`` option.

Reviewer: ```suggestion
To avoid a dependency to be resolved in the workspace, simply close the corresponding project or remove it from the workspace. For the case of Foundation Libraries only, the workspace resolution can also be disabled by unchecking th ``Resolve Foundation Library in workspace`` option in :guilabel:`Window` > :guilabel:`Preferences` > :guilabel:`MicroEJ` > :guilabel:`Settings`.
```
